basic_auth/db/tests/test_model.py

```diff
@@ -57,6 +61,25 @@ class ModelTest(DataBaseTest):
         ]
         self.assertEqual(raw_credentials, expected_credentials)
 
+    async def test_get_all_credentials_date(self):
+        """All credentials can be retrieved."""
+        now = datetime.now()
+        await self.model.add_credentials('user1', 'usernameC', 'pass1')
+
+        expected_credentials = [('user1', 'usernameC', 'pass1')]
+        start = now - timedelta(10)
+        end = now + timedelta(10)
+
+        credentials = await self.model.get_all_credentials(start_date=start)
+        raw_credentials = [
+            (c.user, c.auth.username, c.auth.password) for c in credentials]
+        self.assertEqual(raw_credentials, expected_credentials)
+
+        credentials = await self.model.get_all_credentials(end_date=end)
+        raw_credentials = [
+            (c.user, c.auth.username, c.auth.password) for c in credentials]
+        self.assertEqual(raw_credentials, expected_credentials)
+
```

Comment:
Nice! Could we also add a more high level test to the resource, so that we demonstrate we handle invalid user input etc?